src/builder.ts

```diff
@@ -313,6 +337,29 @@ export class MultiRangeBuilder extends RangeBuilder {
   public exec(): Promise<RPC.IRangeResponse> {
     return this.kv.range(this.request);
   }
+
+  /**
+   * @override
+   */
+  protected createPromise(): Promise<{ [key: string]: string }> {
+    return this.strings();
+  }
+
+  /**
+   * Dispatches a call to the server, and creates a map by running the
+   * iterator over the values returned.
+   */
+  private mapValues<T>(iterator: (buf: Buffer) => T): Promise<{ [key: string]: T }> {
+    return this.exec().then(res => {
+      const output: { [key: string]: T } = {};
+      for (let i = 0; i < res.kvs.length; i += 1) {
+        output[res.kvs[i].key.slice(this.queryPrefix.length).toString()] =
```

Comment:
Yea, it'd make it a little simpler, but I am sort of microoptimizing this a little bit. In a db lib this kind of thing has to potential to get hammered pretty hard.